ClientsListView

vanyalaptii opened the pull request.

vanyalaptii committed:
RepairListDetailView

Added a view with fields for reading, editing information and saving it to database.

vanyalaptii committed:
Fixed UI list update

Was fixed problem when after repair fields changed and back button pressed repairs list not update repair info.

vanyalaptii committed:
Merge branch 'main' into feature/RepairsListDetailsView

vanyalaptii committed:
Merge changes from main

Merge changes from main.

vanyalaptii committed:
ClientsListView, AddNewClientView, ClientDetailView.

1. ClientsListView, ClientsListViewModel added.
2. AddNewClientView, AddNewClientViewModel added.
3. ClientDetailView, ClientDetailViewModel added.
4. ClientManager added.
5. Added properties and coding keys for Client, ConterAgent, UserModel and DBUser models.
6. ProfileView optimized UI.
7. Bug with loadCurrentUser() function that crashes whole app was fixed.
8. Search optimized.

OksanaFedorchuk commented on ReFix/Models/ConterAgent.swift:
```diff
@@ -10,13 +10,18 @@ import Foundation
 class ConterAgent {
```
counter

vanyalaptii replied:
Змінив назву на Counterparty

OksanaFedorchuk commented on ReFix/Modules/Clients/AddNewClientViewModel.swift:
```diff
@@ -0,0 +1,70 @@
+final class AddNewClientViewModel: ObservableObject {
+    
+    @Published private(set) var user: DBUser? = nil
+    @Published var clientDatailIsPresented: Bool = false
```
shouldn't this be in the view?

vanyalaptii replied:
ми переносили цю проперті у VM

OksanaFedorchuk replied:
тобі треба до неї доступ поза в"ю?

OksanaFedorchuk commented on ReFix/Modules/Clients/ClientDetailView.swift:
```diff
@@ -0,0 +1,73 @@
+        VStack {
+            List {
+                customTextField(placeholder: "Ім'я:",text: $viewModel.name)
+                customTextField(placeholder: "Фамілія:",text: $viewModel.surname)
```
Призвіще

vanyalaptii replied:
виправив 😅

OksanaFedorchuk commented on ReFix/Modules/Clients/ClientDetailView.swift:
```diff
@@ -0,0 +1,73 @@
+    }
+}
+
+extension ClientDetailView {
```
private?

vanyalaptii replied:
виправив

OksanaFedorchuk commented on ReFix/Modules/Clients/ClientDetailView.swift:
```diff
@@ -0,0 +1,73 @@
+    }
+}
+
+//#Preview {
```
fix or remove

vanyalaptii replied:
прибрав, але в подальшому буду фіксити

OksanaFedorchuk commented on ReFix/Modules/Clients/ClientDetailViewModel.swift:
```diff
@@ -0,0 +1,62 @@
+    init(client: Binding<Client>) {
+        self.client = client
+        
+        self.id = client.wrappedValue.id
```
func cleanUp() ?

vanyalaptii replied:
змінив назву

OksanaFedorchuk commented on ReFix/Modules/Clients/ClientsListView.swift:
```diff
@@ -0,0 +1,65 @@
+    }
+}
+
+extension ClientsListView {
```
private?

vanyalaptii replied:
виправив

OksanaFedorchuk commented:
Всі правки з попереднього мр, теж актуальні в цьому, поправ будь ласка в обох ці нюанси, щоб я не дублювала коментарі

vanyalaptii committed:
Access control changes

1. Properties and functions marked as private.
2. ConterAgent renamed to Counterparty.

vanyalaptii committed:
Access control

1. Properties and functions marked as private.
2. cleanUp() function renamed to discardUnsavedValues().
3. Search logic moved to searchResult() method.

vanyalaptii committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/RepairsListDetailsView'

Conflicts:
	ReFix.xcodeproj/project.xcworkspace/xcuserdata/vanalaptii.xcuserdatad/UserInterfaceState.xcuserstate
	ReFix/Modules/Repairs/RepairsListView.swift
	ReFix/Modules/Repairs/RepairsListViewModel.swift
	ReFix/Modules/Repairs/RepairsManager.swift

vanyalaptii committed:
Merge changes from feature/RepairsListDetailsView
